lib/filterx/func-str.c

```diff
@@ -497,3 +519,198 @@ filterx_function_includes_new(FilterXFunctionArgs *args, GError **error)
   return _function_affix_new(args, "includes", _function_includes_process, FILTERX_FUNC_INCLUDES_USAGE,
                              error);
 }
+
+static FilterXObject *
+_strcasecmp_eval(FilterXExpr *s)
+{
+  FilterXStrcasecmp *self = (FilterXStrcasecmp *) s;
+  FilterXObject *result = NULL;
+  FilterXObject *a_obj = NULL, *b_obj = NULL;
+
+  const gchar *a_str = NULL;
+  if (!self->a_literal)
+    {
+      a_obj = filterx_expr_eval(self->a.expr);
+      if (!a_obj)
+        goto exit;
+
+      gsize a_str_len = 0;
+      if (!filterx_object_extract_string_ref(a_obj, &a_str, &a_str_len))
+        {
+          filterx_eval_push_error("failed to extract string value", self->a.expr, a_obj);
+          goto exit;
+        }
+    }
+  else
+    a_str = self->a.literal->str;
+
+
+  const gchar *b_str = NULL;
+  if (!self->b_literal)
+    {
+      b_obj = filterx_expr_eval(self->b.expr);
+
+      if (!b_obj)
+        goto exit;
+
+      gsize b_str_len = 0;
+      if (!filterx_object_extract_string_ref(b_obj, &b_str, &b_str_len))
+        {
+          filterx_eval_push_error("failed to extract string value", self->b.expr, b_obj);
+          filterx_object_unref(b_obj);
+          goto exit;
+        }
+    }
+  else
+    b_str = self->b.literal->str;
+
+  /* TODO: utf-8 support together with all func-str-transform */
+  gint cmp = g_ascii_strcasecmp(a_str, b_str);
+  result = filterx_integer_new(cmp);
```

Comment:
we might need to cache integer(-1) otherwise we will be allocating one everytime we return -1 here

Reply:
Yes, it starts from -1: https://github.com/axoflow/axosyslog/blob/main/lib/filterx/object-primitive.c#L420